jans-cedarling/cedarling/src/jwt/decoding_strategy/key_service.rs

```diff
@@ -0,0 +1,137 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+mod error;
+mod openid_config;
+
+pub use error::Error;
+use jsonwebtoken::jwk::JwkSet;
+use jsonwebtoken::DecodingKey;
+use openid_config::*;
+use reqwest::blocking::get;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+pub struct KeyService {
+    idp_configs: HashMap<Box<str>, OpenIdConfig>, // <issuer (`iss`), OpenIdConfig>
```

Comment:
maybe it can be useful to add 
 `client: Client,`
in the struct to use it in `new` and `update_jwks`.

Reply:
resolved by: [7976b58](https://github.com/JanssenProject/jans/pull/9855/commits/7976b5840cfe5ba7851800dc95325b006f1de8ef)